hbase-asyncfs/src/main/java/org/apache/hadoop/hbase/io/asyncfs/FanOutOneBlockAsyncDFSOutput.java

```diff
@@ -231,7 +231,11 @@ private void completed(Channel channel) {
   // so that the implementation will not burn up our brain as there are multiple state changes and
   // checks.
   private synchronized void failed(Channel channel, Supplier<Throwable> errorSupplier) {
-    if (state == State.BROKEN || state == State.CLOSED) {
+    if (state == State.CLOSED) {
+      return;
+    }
+    if (state == State.BROKEN) {
+      failWaitingAckQueue(channel, errorSupplier);
```

Comment:
Good. I think this is the correct way to fix the problem.

Reply:
@Apache9, thank you for the suggestion for the test. Assuming dn2 and dn3 are slow DNs, the simple way I can think  to simulate DNs is to discard the message when flush the dn2 and dn3. Seems it is hard to simulate the slow response from dn2 and dn3, because it seems to require hack to Netty implementation and seems more complex, and because Netty event loop is a single thread, I could not block in it to hang other messages. What is your opinion?

Reply:
Let me have a try. Will report back here to see if it is possible.

Reply:
@Apache9 ,thank you very much, in any case , my original test using reflection to create MyFanOutOneBlockAsyncDFSOutput is not proper.I have improved my test to remove reflection and use  Mockito to simulate slow DNs  to discard the message when flushing to dn2 and dn3. FYI.